### PR TITLE
Protobuff refacto

### PR DESCRIPTION
**Pull request by antoine-de**
the disruption test do not pass yet (a certain 'throw TODO' in the protobuff), but now it builds :smile:

### CODE REVIEW

File: source/disruption/tests/disruption_test.cpp [hunk 1]
```diff
@@ -56,98 +56,156 @@ end_application_date    = 2014-02-02 18:32:00
...
+};
+
+class Params {
 public:
     std::vector<std::string> forbidden;
     ed::builder b;
     size_t period;
 
-    Params():b("20120614"), period(365){
+    void add_disruption(disruption_creator disrupt,nt::PT_Data& pt_data) {
```
Comment (TeXitoi) [inline source/disruption/tests/disruption_test.cpp:105]: you can take disrupt as const reference, but we don't really care.

File: source/type/datetime.h [hunk 2]
```diff
@@ -173,4 +174,13 @@ inline weekdays get_weekday(const boost::gregorian::date& date) {
...
 
+/*
+ * Split a period starting from 'start' to 'end'
+ *
+ * the period is split for each day, each day starting from 'beg_of_day' to 'end_of_day'
+ */
+std::vector<boost::posix_time::time_period>
+split_period(boost::posix_time::ptime start, boost::posix_time::ptime end,
+             boost::posix_time::time_duration beg_of_day, boost::posix_time::time_duration end_of_day,
+             std::bitset<7> days);
```
Comment (TeXitoi) [inline source/type/datetime.h:185]: I find the name quite strange, but I don't have better proposition (expand_calendar?)

File: source/type/datetime.cpp [hunk 3]
```diff
@@ -86,4 +86,38 @@ boost::posix_time::ptime from_posix_timestamp(uint64_t val) {
...
+             boost::posix_time::time_duration beg_of_day, boost::posix_time::time_duration end_of_day,
+             std::bitset<7> days) {
+    if (days.all() && beg_of_day == boost::posix_time::hours(0) && end_of_day == boost::posix_time::hours(24)) {
+        return {{start, end}};
+    }
+    auto period = boost::gregorian::date_period(start.date(), end.date());
+
+    std::vector<boost::posix_time::time_period> res;
+    for(boost::gregorian::day_iterator it(period.begin()); it < period.end(); ++it) {
+        auto day = (*it);
```
Comment (TeXitoi) [inline source/type/datetime.cpp:103]: Can't you just `for (const auto& day: period) {` ?

File: source/type/datetime.cpp [hunk 4]
```diff
@@ -86,4 +86,38 @@ boost::posix_time::ptime from_posix_timestamp(uint64_t val) {
...
+        //since monday is the first day of the bitset, we need to convert the weekday to a 'french' week day
+        navitia::weekdays week_day = navitia::get_weekday(day);
+        if(! days.test(week_day)) {
+            continue;
+        }
+        res.push_back(boost::posix_time::time_period(boost::posix_time::ptime(day, beg_of_day), boost::posix_time::ptime(day, end_of_day)));
```
Comment (TeXitoi) [inline source/type/datetime.cpp]: long line (fixable with something like `using namespace boost::posix_time`)

File: source/type/tests/datetime.cpp [hunk 5]
```diff
@@ -214,3 +214,45 @@ BOOST_AUTO_TEST_CASE(time_dur_overflow) {
...
+
+    BOOST_REQUIRE_EQUAL(periods.size(), 8);
+
+    //first starts from 12:00 (because it's beg's time) and finish at 14:00 (it's the regular end of a day)
+    BOOST_CHECK_EQUAL(periods[0].begin(), boost::posix_time::from_iso_string("20140101T120000"));
+    BOOST_CHECK_EQUAL(periods[0].end(), boost::posix_time::from_iso_string("20140101T140000"));
+
+    for (int i = 1; i < 7 ; ++i) {
+        BOOST_CHECK_EQUAL(periods[i].begin(), pt::ptime(boost::gregorian::date(2014,01,i+1), pt::duration_from_string("11:00")));
+        BOOST_CHECK_EQUAL(periods[i].end(), pt::ptime(boost::gregorian::date(2014,01,i+1), pt::duration_from_string("14:00")));
```
Comment (TeXitoi) [inline source/type/tests/datetime.cpp]: 2 long lines

File: source/disruption/tests/disruption_test.cpp [hunk 6]
```diff
@@ -56,98 +56,156 @@ end_application_date    = 2014-02-02 18:32:00
...
-        b.vj("network:Test","line:test","11111111","",true, "")("stop_area:stop22", 8*3600 +10*60, 8*3600 + 11 * 60)("stop_area:stop22", 8*3600 + 20 * 60 ,8*3600 + 21*60);
+        b.vj("network:R", "line:A", "11111111", "", true, "")("stop_area:stop1", 8*3600 +10*60, 8*3600 + 11 * 60)
+                ("stop_area:stop2", 8*3600 + 20 * 60 ,8*3600 + 21*60);
+        b.vj("network:R", "line:S", "11111111", "", true, "")("stop_area:stop5", 8*3600 +10*60, 8*3600 + 11 * 60)
+                ("stop_area:stop6", 8*3600 + 20 * 60 ,8*3600 + 21*60);
+        b.vj("network:K","line:B","11111111","",true, "")("stop_area:stop3", 8*3600 +10*60, 8*3600 + 11 * 60)
+                ("stop_area:stop4", 8*3600 + 20 * 60 ,8*3600 + 21*60);
+        b.vj("network:M","line:M","11111111","",true, "")("stop_area:stop22", 8*3600 +10*60, 8*3600 + 11 * 60)
+                ("stop_area:stop22", 8*3600 + 20 * 60 ,8*3600 + 21*60);
+        b.vj("network:Test","line:test","11111111","",true, "")("stop_area:stop22", 8*3600 +10*60, 8*3600 + 11 * 60)
```
Comment (TeXitoi) [inline source/disruption/tests/disruption_test.cpp:150]: I didn't see it's a long line (by 2 char)

File: source/type/datetime.cpp (hunk 3 above)
Comment (antoine-de) [inline source/type/datetime.cpp:103]: nop begin() returns a date, not an iterator :/

**Comment by antoine-de** [comment]
@TeXitoi code updated

**Pull request merged**